Add migration to charts4 chapter

rogozinds opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/charts/353)
<!-- Reviewable:end -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/vaadin/charts?pullRequest=353) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/vaadin/charts?pullRequest=353) before we can accept your contribution.<br/>

alvarezguille commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 22 at r1](https://reviewable.io:443/reviews/vaadin/charts/353#-KWCP9lUmsOrRNYeoPkt:-KWCP9lUmsOrRNYeoPku:bu9rh0r) ([raw file](https://github.com/vaadin/charts/blob/c41ea420ef75e20ecb6fd26181433b2bcd54a3b1/documentation/java-api/charts-migration-vaaadin8.asciidoc#L22)):*
> Methods that use [classname]#Date# marked as deprecated
were deprecated

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 23 at r1](https://reviewable.io:443/reviews/vaadin/charts/353#-KWCPIm9RDpsrsxYqobM:-KWCPIm9RDpsrsxYqobN:b-2udgw8) ([raw file](https://github.com/vaadin/charts/blob/c41ea420ef75e20ecb6fd26181433b2bcd54a3b1/documentation/java-api/charts-migration-vaaadin8.asciidoc#L23)):*
> * [classname]#ContainerDataSeries# is removed, use [classname]#ChartDataSeries# instead, see <<charts-data#charts.data.dataseries,"instructions">>.
It should mention that there's a compatibility package if ContainerDataSeries is needed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/charts/353)*
<!-- Sent from Reviewable.io -->

rogozinds commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions.

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 22 at r1](https://reviewable.io:443/reviews/vaadin/charts/353#-KWCP9lUmsOrRNYeoPkt:-KWYAuqK3qGLDZnyLjMy:b-896fix) ([raw file](https://github.com/vaadin/charts/blob/c41ea420ef75e20ecb6fd26181433b2bcd54a3b1/documentation/java-api/charts-migration-vaaadin8.asciidoc#L22)):*
<details><summary><i>Previously, alvarezguille (Guille) wrote…</i></summary>
> > Methods that use [classname]#Date# marked as deprecated
> were deprecated
</details>
Done.

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 23 at r1](https://reviewable.io:443/reviews/vaadin/charts/353#-KWCPIm9RDpsrsxYqobM:-KWYFG2jaGWsIOeerULp:b-896fix) ([raw file](https://github.com/vaadin/charts/blob/c41ea420ef75e20ecb6fd26181433b2bcd54a3b1/documentation/java-api/charts-migration-vaaadin8.asciidoc#L23)):*
<details><summary><i>Previously, alvarezguille (Guille) wrote…</i></summary>
> > * [classname]#ContainerDataSeries# is removed, use [classname]#ChartDataSeries# instead, see <<charts-data#charts.data.dataseries,"instructions">>.
> It should mention that there's a compatibility package if ContainerDataSeries is needed
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/charts/353)*
<!-- Sent from Reviewable.io -->

alvarezguille commented:
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/charts/353)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: all files reviewed at latest revision, 7 unresolved discussions, all commit checks successful.

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 13 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWb_Y2rai9rOz7h4KE5:-KWb_Y2rai9rOz7h4KE6:b-az8c5u) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L13)):*
> ```AsciiDoc
> == Introduction
> 
> Charts 4 is using Vaadin 8.
> ```

Vaadin Framework 8

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 14 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWb_wDm0YnvkU8rJt1-:-KWb_wDm0YnvkU8rJt10:byjx3pr) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L14)):*
> ```AsciiDoc
> 
> Charts 4 is using Vaadin 8.
> Before migrating from Charts 3 to Charts 4, you need to migrate your application to Vaadin 8, see <<dummy/../../../charts/java-api/charts-configuration#charts.configuration,"instructions">>.
> ```

Is this the correct link?

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 22 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWb_yO4OBvpR9nkNvjL:-KWb_yO4OBvpR9nkNvjM:b-9af1hq) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L22)):*
> ```AsciiDoc
> There are two main changes:
> 
> * Methods that use [classname]#Date# were deprecated, use [classname]#LocalDate# instead.
> ```

There is no API using LocalDate?

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 23 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWb_iFbzICKL7SnJK87:-KWb_iFbzICKL7SnJK88:b-cn3hsp) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L23)):*
> ```AsciiDoc
> 
> * Methods that use [classname]#Date# were deprecated, use [classname]#LocalDate# instead.
> * [classname]#ContainerDataSeries# is removed, use [classname]#ChartDataSeries# instead, see <<charts-data#charts.data.dataseries,"instructions">>.
> ```

ContainerDataSeries is still there in the compatibility package - it was not removed

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 23 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWb_rYzos8GhGE4TZm_:-KWb_rYzos8GhGE4TZma:b-nurdwl) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L23)):*
> ```AsciiDoc
> 
> * Methods that use [classname]#Date# were deprecated, use [classname]#LocalDate# instead.
> * [classname]#ContainerDataSeries# is removed, use [classname]#ChartDataSeries# instead, see <<charts-data#charts.data.dataseries,"instructions">>.
> ```

ChartDataSeries -> DataProviderSeries

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 29 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWbaMYBvqEjfjwLbogy:-KWbaMYBvqEjfjwLbogz:b-13vcra) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L29)):*
> ```AsciiDoc
> 
> During migration from Charts 3 to Charts 4 you may want to use [classname]#ContainerDataSeries#.
> To do that, add a dependency to vaadin-charts-compatibility package:
> ```

... dependency to the

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 40 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWbaUjj8RLdh69Rl2zR:-KWbaUjj8RLdh69Rl2zS:b44bqlm) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L40)):*
> ```AsciiDoc
> ----
> 
> Afterwards, change import statement:
> ```

change the import statement for `ContainerDateSeries` to

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/charts/353)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 45 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWbanPr9Q3wgbWTQDtb:-KWbanPsw2noFx3b87Pk:b-4yr6dt) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L45)):*
> ```AsciiDoc
> ----
> import com.vaadin.v7.addon.charts.model.ContainerDataSeries;
> ----
> ```

newline

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/charts/353)*
<!-- Sent from Reviewable.io -->

rogozinds commented:
Review status: all files reviewed at latest revision, 8 unresolved discussions.

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 13 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWb_Y2rai9rOz7h4KE5:-KWbbjPiYG7jI-tx8_Du:b-896fix) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L13)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary>
> Vaadin Framework 8
</details>
Done.

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 14 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWb_wDm0YnvkU8rJt1-:-KWbbQ7qJfUL-eRH8NZT:b-nm00oq) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L14)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary>
> Is this the correct link?
</details>
Nope. There is no "correct" link, because there is no vaadin 8 documentation.

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 22 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWb_yO4OBvpR9nkNvjL:-KWbcBiBRpMXrlB_wW-f:buqko9t) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L22)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary>
> There is no API using LocalDate?
</details>
Changed to Instant.

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 23 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWb_iFbzICKL7SnJK87:-KWbuKFIYFD9iBC43tSo:b-dbvp4p) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L23)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary>
> ContainerDataSeries is still there in the compatibility package - it was not removed
</details>
Changed to deprecated

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 23 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWb_rYzos8GhGE4TZm_:-KWbtY2-AQJqRk4Ol_dj:b-896fix) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L23)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary>
> ChartDataSeries -> DataProviderSeries
</details>
Done.

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 29 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWbaMYBvqEjfjwLbogy:-KWbta1tEmJh_-tNezYe:bs7wumb) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L29)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary>
> ... dependency to the
</details>
Done.

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 40 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWbaUjj8RLdh69Rl2zR:-KWbuCWmnay9f9Rg881x:b-896fix) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L40)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary>
> change the import statement for `ContainerDateSeries` to
</details>
Done.

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 45 at r2](https://reviewable.io:443/reviews/vaadin/charts/353#-KWbanPr9Q3wgbWTQDtb:-KWbu-tN9dPbmHxSEad-:b-896fix) ([raw file](https://github.com/vaadin/charts/blob/f8405f00297841858ec4c853e77f435bedbb7ab8/documentation/java-api/charts-migration-vaaadin8.asciidoc#L45)):*
<details><summary><i>Previously, Artur- (Artur) wrote…</i></summary>
> newline
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/charts/353)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[documentation/java-api/charts-migration-vaaadin8.asciidoc, line 40 at r3](https://reviewable.io:443/reviews/vaadin/charts/353#-KWc2alXqscIxSYqTwYz:-KWc2alXqscIxSYqTwZ-:b-who92f) ([raw file](https://github.com/vaadin/charts/blob/62ac5d84715ce1b428f78917f475e7ac01fc409a/documentation/java-api/charts-migration-vaaadin8.asciidoc#L40)):*
> ```AsciiDoc
> ----
> 
> Afterwards, change the import statement for ContainerDateSeries to:
> ```

`ContainerDataSeries`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/charts/353)*
<!-- Sent from Reviewable.io -->

Artur- commented:
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/vaadin/charts/353)*
<!-- Sent from Reviewable.io -->
